landoapi/hgexports.py

```diff
@@ -71,12 +75,92 @@ def _no_line_breaks(break_string: str) -> str:
     return "".join(break_string.strip().splitlines())
 
 
+def parse_git_author_information(user_header: bytes) -> tuple[bytes, bytes]:
+    """Parse user's name and email address from a Git style author header.
+
+    Converts a header like 'User Name <user@example.com>' to its separate parts.
+    """
+    name, email = parseaddr(user_header.decode("utf-8"))
+    return name.encode("utf-8"), email.encode("utf-8")
+
+
+def get_timestamp_from_git_date_header(date_header: bytes) -> str:
+    """Convert a Git patch date header into a timestamp."""
+    header_datetime = dateparse(date_header)
+    return str(math.floor(header_datetime.timestamp()))
+
+
+def get_timestamp_from_hg_date_header(date_header: bytes) -> bytes:
+    """Return the first part of the `hg export` date header.
+
+    >>> get_timestamp_from_hg_date_header(b"1686621879 14400")
+    b"1686621879"
+    """
+    return date_header.split(b" ")[0]
+
+
 class PatchHelper(object):
-    """Helper class for parsing Mercurial patches/exports."""
+    """Base class for parsing patches/exports."""
 
     def __init__(self, fileobj: io.BytesIO):
         self.patch = fileobj
         self.headers = {}
+
+    @staticmethod
+    def _is_diff_line(line: bytes) -> bool:
+        return DIFF_LINE_RE.search(line) is not None
+
+    def header(self, name: bytes | str) -> Optional[bytes]:
```

Comment:
Might be better to rename this to `get_header` to make things a little more consistent.